### PR TITLE
Vote security & fixes

### PR DESCRIPTION
**Pull request by cryptoseneca**
- Fixes bug where non validated votes where being counted as valid
- Add replay prevention
- Removes `signatureState` from payload to be signed

### CODE REVIEW

File: packages/prop-house-backend/src/proposal/proposal.entity.ts [hunk 1]
```diff
@@ -58,6 +59,7 @@ export class Proposal extends SignedEntity {
   @BeforeUpdate()
   updateVoteCount() {
     this.voteCount = this.votes.reduce((acc, vote) => {
+      if (vote.signatureState !== SignatureState.VALIDATED) return;
```
Comment (solimander) [inline packages/prop-house-backend/src/proposal/proposal.entity.ts:62]: Should be `if (vote.signatureState !== SignatureState.VALIDATED) return acc;`, right?

File: packages/prop-house-backend/src/vote/vote.entity.ts [hunk 2]
```diff
@@ -41,6 +41,10 @@ export class Vote extends SignedEntity {
   @Field(() => Int)
   weight: number;
 
+  @Column({ default: null })
+  @Field(() => Int)
+  blockHeight: number;
```
Comment (solimander) [inline packages/prop-house-backend/src/vote/vote.entity.ts:46]: Missing a migration to add this column

File: packages/prop-house-wrapper/src/builders.ts [hunk 3]
```diff
@@ -149,11 +137,65 @@ export class Vote extends Signable {
...
+    signer: Signer | Wallet,
+    jsonPayload: string,
+    signature: string,
+  ) {
+    const address = await signer.getAddress();
+    return {
+      signedData: {
+        message: Buffer.from(jsonPayload).toString('base64'),
+        signature,
+        signer: address,
```
Comment (solimander) [inline packages/prop-house-wrapper/src/builders.ts:170]: Doesn't really matter for this PR, but want to point out that `signedData.signer` is unnecessary.

File: packages/prop-house-backend/src/proposal/proposal.entity.ts (hunk 1 above)
Comment (cryptoseneca) [reply packages/prop-house-backend/src/proposal/proposal.entity.ts:62]: yep! thanks for catching.

**Pull request merged**